lib/spack/spack/cmd/ci.py

```diff
@@ -816,6 +835,167 @@ def _gitlab_artifacts_url(url: str) -> str:
     return urlunparse(parsed._replace(path="/".join(parts), fragment="", query=""))
 
 
+def validate_standard_versions(pkg, versions: VersionList) -> bool:
+    """Get and test the checksum of a package version based on a tarball.
+
+    Args:
+      pkg (spack.package_base.PackageBase): Spack package for which to validate a version checksum
+      versions (spack.version.VersionList): list of package versions to validate
+
+    Returns: (bool): result of the validation. True is valid and false is failed.
+    """
+    url_dict: Dict[StandardVersion, str] = {}
+
+    for version in versions:
+        url = pkg.find_valid_url_for_version(version)
+        url_dict[version] = url
+
+    version_hashes = spack.stage.get_checksums_for_versions(
+        url_dict, pkg.name, fetch_options=pkg.fetch_options
+    )
+
+    valid_checksums = True
+    for version, sha in version_hashes.items():
+        if sha != pkg.versions[version]["sha256"]:
+            tty.error(
+                f"Invalid checksum found {pkg.name}@{version}\n"
+                f"    [package.py] {pkg.versions[version]['sha256']}\n"
+                f"    [Downloaded] {sha}"
+            )
+            valid_checksums = False
+
+        tty.info(f"Validated {pkg.name}@{version} --> {sha}")
+
+    return valid_checksums
+
+
+def validate_git_versions(pkg, versions: VersionList) -> bool:
+    """Get and test the commit and tag of a package version based on a git repository.
+
+    Args:
+      pkg (spack.package_base.PackageBase): Spack package for which to validate a version
+      versions (spack.version.VersionList): list of package versions to validate
+
+    Returns: (bool): result of the validation. True is valid and false is failed.
+    """
+    git = spack.util.git.git(required=True)
+
+    with tempfile.TemporaryDirectory() as tmpdirpath:
+        # Test if repository can be cloned.
+        try:
+            git("clone", pkg.git, tmpdirpath, output=str, error=str)
```

Comment:
No, but if you don't redirect `stdout/err` to a string they print to the CI job's output and make it difficult to scan for valid / invalid checksums in the output since you've got the verbose git output in there.

Reply:
The standard way of acknowledging a function return value that will not be used is to assign it to `_`, so `_ = git(...`.  Not sure how others on the team feel about that but IMO it is useful for ppl unfamiliar with Spack.